otx/algorithms/common/adapters/mmcv/tasks/stage.py

```diff
@@ -486,12 +487,34 @@ def get_model_classes(cfg):
     def get_model_ckpt(ckpt_path, new_path=None):
         """Return model ckpt from ckpt_path."""
         ckpt = CheckpointLoader.load_checkpoint(ckpt_path, map_location="cpu")
+        local_torch_hub_folder = torch.hub.get_dir()
         if "model" in ckpt:
             ckpt = ckpt["model"]
             if not new_path:
                 new_path = ckpt_path[:-3] + "converted.pth"
-            torch.save(ckpt, new_path)
+            if not torch.distributed.is_initialized() or dist.get_rank() == 0:
+                torch.save(ckpt, new_path)
             return new_path
+
+        elif "state_dict" in ckpt:
+            ckpt = ckpt["state_dict"]
+            new_ckpt = OrderedDict()
+            modified = False
+            # patch pre-trained checkpoint for model
+            for name in ckpt:
+                if not name.startswith("backbone") and not "fc" in name and not "head" in name:
```

Comment:
Could you add a comment in code to describe why it's necessary?

Reply:
Sure, I will add comment in the code